python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/utils.py

```diff
@@ -79,11 +77,34 @@ def _resolve_p2p_ips(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol, p
             ip_pool = self.inputs_data.p2p_links_ip_pools[p2p_link.ip_pool]
             if not ip_pool.ipv4_pool:
                 # The pool was missing ipv4_pool so we give up.
-                return p2p_link
+                pass
+            else:
+                p2p_link.ip.extend(
+                    [
+                        f"{get_ip_from_pool(ip_pool.ipv4_pool, ip_pool.prefix_size, p2p_link.id - 1, host_offset)}/{ip_pool.prefix_size}"
+                        for host_offset in [0, 1]
+                    ]
+                )
 
-            p2p_link.ip.extend(
-                [f"{get_ip_from_pool(ip_pool.ipv4_pool, ip_pool.prefix_size, p2p_link.id - 1, host_offset)}/{ip_pool.prefix_size}" for host_offset in [0, 1]]
-            )
+        if self.shared_utils.underlay_ipv6 is True:
```

Comment:
This condition means we silently ignore the ipv6 stuff unless you enable ipv6_underlay. That seems a bit weird. I know I asked you about this kind of check, but I think we need to either raise or not check this and instead just configure ipv6 routing. (so not enable the full underlay, but just ensure that these interface IPs actually work)

```suggestion
        if self.shared_utils.underlay_ipv6:
```

Reply:
I opted to remove the requirement and add ipv6 routing to the config if there are ipv6 enabled links.